### ISSUE TITLE
In ACCT_IndividualAccounts_TDTM line 248, Too many query rows: 50001 when account has >50000 contacts

### ISSUE THREAD

**Issue by vegad** [SalesforceFoundation/HEDAP#425]
There is a line of code that is checking for a new primary contact (line 248) in:
https://github.com/SalesforceFoundation/HEDAP/blob/02b9f382bccb8d56d9e7145a0c61a3bc88f97d41/src/classes/ACCT_IndividualAccounts_TDTM.cls
~~~~
        list<Contact> listCon = [select Id, AccountId, Account.Primary_Contact__c from Contact where AccountId in :listAccountId];
~~~~
For some context, here is the method and some surrounding code:
~~~
 private void resetPrimaryContactForAccount(list<Id> listAccountId, DmlWrapper dmlWrapper) {
    
        // for each account, we need a set of their contacts.
        map<Id, set<Id>> mapAccountIdSetContacts = new map<Id, set<Id>>();
        
        list<Contact> listCon = [select Id, AccountId, Account.Primary_Contact__c from Contact where AccountId in :listAccountId];
        for (Contact con : listCon) {
            set<Id> setContactId = mapAccountIdSetContacts.get(con.AccountId);
            if (setContactId == null) { 
                setContactId = new set<Id>();
                mapAccountIdSetContacts.put(con.AccountId, setContactId);
            }
            setContactId.add(con.Id);
        }
~~~

This line queries the account for all of its contacts.  When that account has > 50000 contacts, as they sometimes due when using a "bucket" account model, this throws an exception. Too many query rows: 50001

Please consider limiting the query so that less contacts are returned.  

Currently, I have a workaround using the filters to exclude this bucket account from the trigger, but that likely wont last forever.  

Thanks!

**Comment by davidjray** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000006PJd8AAG" target="_blank">W-018674</a>

**Comment by jacebryan** [comment]
Thank you @vegad for submitting this issue.

**Comment by mrbelvedere** [comment]
Included in beta release 1.36 (Beta 9)

**Comment by mrbelvedere** [comment]
Included in production release rel/1.36
